File: ios/App/App/UpdateCheckerPlugin.m
#import <Capacitor/Capacitor.h>

// Capacitor si pluginy hledá přes Objective-C runtime, ne přes Swift import —
// bez tohohle registračního souboru by JS strana (Capacitor.Plugins.UpdateChecker)
// o UpdateCheckerPlugin.swift vůbec nevěděla.
CAP_PLUGIN(UpdateCheckerPlugin, "UpdateChecker",
    CAP_PLUGIN_METHOD(checkNow, CAPPluginReturnPromise);
)

File: ios/App/App/UpdateCheckerPlugin.swift
import Foundation
import Capacitor

/// Most mezi JS a UpdateCheckerem: tlačítko "Zkontrolovat aktualizace" ve
/// webu volá tohle, aby obešlo šestihodinovou pojistku v
/// UpdateChecker.checkAndPrompt (viz komentář tam) — bez ní by uživatel
/// musel čekat na další automatickou kontrolu při přepnutí do appky.
@objc(UpdateCheckerPlugin)
public class UpdateCheckerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "UpdateCheckerPlugin"
    public let jsName = "UpdateChecker"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "checkNow", returnType: CAPPluginReturnPromise)
    ]

    /// Spustí kontrolu a rovnou vrátí JS zpátky — výsledek (nová verze /
    /// "máš nejnovější" / chyba) appka zobrazí sama nativním dialogem,
    /// není co posílat zpátky do webu.
    @objc func checkNow(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            UpdateChecker.checkAndPrompt(force: true)
        }
        call.resolve()
    }
}
